src/ssh_config.rs

```diff
@@ -106,7 +128,7 @@ fn find_ssh_key_for_host_in_config(
                             ))
                         })?;
 
-                    if key.as_str().eq_ignore_ascii_case("IdentityFile") {
+                    if key.as_str().eq_ignore_ascii_case(name) {
                         let path = value.as_str().to_string();
 
                         // trace!("found IdentityFile option with value {:?}", path);
```

Comment:
As you modify the function (and this part) to not only retrieve a path of IdentityFile, can you also rename the variable `path` to reflect this change eg `found_value`).

Maybe in the future, an optimisation will be to not parse the file for every search.

Reply:
fixed by c140c6794874453edc47b964f947ddd9c7b367d7